SSH Auth docs improvement

k0dard opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

Some small changes in the ssh auth part of the docs:
- Some commands concerning generating ssh keys were misleading (cat is used for viewing files and not generating ssh keys)
- Added formatting for ssh-privatekey and known-hosts keys
- Added instructions on how to get known-hosts value for GitHub

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

k0dard committed:
Update auth.md

More precise instructions on how to setup git ssh key secret

k0dard committed:
Update auth.md

k0dard committed:
Update auth.md

Formatting improvements

k0dard commented:
/kind documentation

vdemeester commented on docs/auth.md:
```diff
@@ -236,7 +236,11 @@ This section describes how to configure an `ssh-auth` type `Secret` for use with
-1. In `secret.yaml`, define a `Secret` that specifies your SSH private key:
+1. Generate a new SSH key with `ssh-keygen` command (optionally specify the file path in which the key will be saved, default is `~/.ssh/<key-name>`)
+
+2. View public key with `cat ~/.ssh/<key-name>.pub` and add it to GitHub (under Settings > SSH and GPG Keys > New SSH key)
```
This is an optional step though (one could use something else than GitHub)

vdemeester reviewed (commented):
One comment.
The `1.` are so that we don't need to update all the numbers if we add one new item (it's a markdown feature 😛 )

vdemeester committed:
docs: use auto-numbered ordered list in ssh-auth section

Markdown auto-numbers list items written as '1.', so the steps no longer
need to be renumbered when items are added or removed. The explicit
numbering had also skipped step 4, rendering as 1,2,3,5,6,7.

vdemeester committed:
docs: align comment wording with known_hosts key name

vdemeester commented on docs/auth.md:
```diff
@@ -247,23 +251,26 @@ specified in the `Secret`.
      # If it is not provided then the git server's public key will be requested
      # when the repo is first fetched.
-     known_hosts: <known-hosts>
+     known_hosts: |
```
Good catch — fixed in 1d81013, the comment now uses `known_hosts` to match the key.

vdemeester reviewed (approved):
/lgtm
